presto-hive-common/src/test/java/com/facebook/presto/hive/zorder/TestZOrderIntegers.java

```diff
@@ -0,0 +1,223 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.zorder;
+
+import com.google.common.collect.ImmutableList;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+import static com.google.common.base.Strings.nullToEmpty;
+import static java.lang.String.format;
+import static org.testng.Assert.fail;
+
+public class TestZOrderIntegers
+{
+    @Test
+    public void testZOrderNegativeIntegers()
+    {
+        ZOrder zOrder = new ZOrder(ImmutableList.of(8, 8, 8, 8, 8, 8));
+        List<Integer> intColumns = ImmutableList.of(1, 2, 3, -10, 4, 5);
+
+        try {
+            zOrder.encodeToByteArray(intColumns);
+            fail("Expected test to fail: z-ordering does not support negative integers.");
+        }
+        catch (IllegalArgumentException e) {
+            String expectedMessage = "Our current Z-Ordering implementation does not support negative input numbers.";
+            if (!nullToEmpty(e.getMessage()).equals(expectedMessage)) {
```

Comment:
nit: How about using `assertEquals`? This could also apply to other places in the tests when comparing strings and lists.